src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/internal/JDBCQueryHelper.java

```diff
@@ -0,0 +1,568 @@
+            for(Field<?> field:fields) {
+                if(i++>0) builder.append(", ");
+                builder.append(field.fieldExpression);
+            }
+        }
+        builder.append(" FROM " + table + " WHERE ");
+        sel.appendCondition(builder);
+        builder.append(";");
+        
+        return builder;
+    }
+    
+    public Map<String, Object> selectQuery(String table, Selector sel, Field<?>... fields) {
+        return anyQuery(createSelect(table, sel, fields), fields);     
+    }
+
+    public List<Map<String, Object>> multiSelectQuery(String table, Selector sel, Field<?>... fields) {
+        return anyMultiQuery(createSelect(table, sel, fields), fields);
+    }
+    
+    public InputStream blobQuery(String table, Selector sel, Field<InputStream> field) {
+        return anyBlobQuery(createSelect(table, sel, field), field);
+    }
+    
+    public int deleteQuery(String table, Selector sel) {
+        QueryBuilder builder = new QueryBuilder();
+        
+        builder.append("DELETE FROM " + table + " WHERE ");
+        sel.appendCondition(builder);
+        builder.append(";");
+        
+        return anyUpdateQuery(builder);
+    }
+    
+    public int updateQuery(String table, Selector sel, Assignment<?>... assignments) {
+        QueryBuilder builder = new QueryBuilder();
+        
+        builder.append("UPDATE " + table + " SET ");
+        {
+            int i=0;
+            for(Assignment<?> assign:assignments) {
+                if(i++>0) builder.append(", ");
+                assign.appendAssignment(builder);
+            }
+        }
+        builder.append(" WHERE ");
+        sel.appendCondition(builder);
+        builder.append(";");
+        
+        return anyUpdateQuery(builder);
+    }
+    
+    public Integer insertQuery(String table, Assignment<?>... assignments) {
+        QueryBuilder builder = new QueryBuilder();
+        
+        builder.append("INSERT INTO " + table + "( ");
+        {
+            int i=0;
+            for(Assignment<?> assign:assignments) {
+                if(i++>0) builder.append(", ");
+                builder.append(assign.getField().getFieldName());
+            }
+        }        
+        builder.append(") VALUES ( ");
+        {
+            int i=0;
+            for(Assignment<?> assign:assignments) {
+                if(i++>0) builder.append(", ");
+                builder.append("?");
+                assign.addAsParameter(builder);
+            }
+        }
+        builder.append(");");
+        
+        List<Integer> result = anyInsertQuery(builder);
+        assert(result == null || result.size() == 1);
+        return result==null ? null : result.get(0);
+    }
+        
+    public List<Map<String, Object>> anyMultiQuery(QueryBuilder query, Field<?>... fields) {
+        try (Connection c = ds.getConnection()) {
+            LOGGER.log(Level.FINEST, query.toString());
+            PreparedStatement stmt = query.toStatement(c);
```

Comment:
Yeah I updated the Connection blocks just minutes before creating this PR (because I could look in the future and I ask me this change) but I forgot about the statements and Resultsets, doh!